abci/client/committing_client.go

```diff
@@ -21,7 +21,8 @@ type committingClient struct {
 	// InitChain
 	// Commit
 	// ApplySnapshotChunk
-	mtx *tmsync.RWMutex
+	deliverMtx *tmsync.RWMutex
+	mtx        *tmsync.RWMutex
```

Comment:
The comment that previously applied to `mtx` now seems to apply to `deliverMtx`, leaving `mtx` undocumented. Please document each with the expectations of who should take which kind of lock and when. Especially document their eventual co-reference.